### PR TITLE
fix: Allow modification of stack values by using references

### PR DESCRIPTION
**Pull request by kattrali**
Makes it possible to modify stack frame values without having to
completely reconstruct the stack from modified frames.

For example, this now works:

```php
\Bugsnag::registerCallback(function ($report) {
    $frames = &$report->getStacktrace()->getFrames();
    foreach ($frames as &$frame) {
        $frame["inProject"] = true;
    }
});
```

## Changeset

Added reference indicators to `toArray()` and `getFrames()`

## Tests

* Added a new callback test which changes a frame value by reference

## Review

- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: tests/ClientTest.php [hunk 1]
```diff
@@ -159,6 +159,28 @@ public function testBeforeNotifySkipsError()
...
+
+        $this->client->setBatchSending(false);
+
+        $this->client->notify($report = Report::fromNamedError($this->config, 'Magic', 'oh no'));
+
+        $this->assertFalse($report->getStacktrace()->getFrames()[0]['inProject']);
+
+        $this->client->registerCallback(function (Report $report) {
+            $frames = &$report->getStacktrace()->getFrames();
+            $frames[0]['inProject'] = true;
```
Comment (Cawllec) [inline tests/ClientTest.php]: Can we test against something that explicitly won't be there, as `inProject` could conceivably be `true`, making this a false positive (or at least ambiguous to read).  Something like:
`$frames[0]['testFlag'] = true;`
With a corresponding assertion?

Comment (kattrali) [reply tests/ClientTest.php]: That's why there are two assertions in this test. One to check that the value is false by default and then adding a callback, notifying, and checking that the value is true.

**Comment by GrahamCampbell** [comment]
This might lead to unintended side effects if people are modifying the result from the getter.

**Comment by kattrali** [comment]
Don’t really see how in a practical sense; it would still need to be referenced correctly to modify the report.
